js/menuRenderer.js

```diff
@@ -94,6 +95,58 @@ module.exports = {
       browserUI.addTab(newTab, {
         enterEditMode: !data.url // only enter edit mode if the new tab is empty
       })
+
+      if (data.taskQuery) {
+        // use the first search result
+        // if there is no result, need to create a new task
+        let task
+
+        if (/^\d+$/.test(data.taskQuery)) {
+          task = tasks.get(data.taskQuery)
+        } else {
+          task = searchAndSortTasks(data.taskQuery, excludeSelected=false)[0]?.task
+        }
+
+        if (!task) {
+          task = tasks.get(tasks.add(undefined, tasks.getIndex(tasks.getSelected().id) + 1))
+          task.name = data.taskQuery
+        }
+
+        moveToTaskCommand(task.id)
+      }
+      
+    })
+
+    ipc.on('switchToTask', function (e, data) {
+      /* new tabs can't be created in modal mode */
+      if (modalMode.enabled()) {
+        return
+      }
+
+      /* new tabs can't be created in focus mode */
+      if (focusMode.enabled()) {
+        focusMode.warn()
+        return
+      }
+
+      if (data.taskQuery) {
+        // use the first search result
+        // if there is no result, need to create a new task
+        let task
+
+        if (/^\d+$/.test(data.taskQuery)) {
```

Comment:
Maybe this would be simpler / better as:
```
if (tasks.get(data.taskQuery)) {
    task = tasks.get...
} else {
    task = searchAndSortTasks...
}

if (!task) {
    task = tasks.get(tasks.add(...
}
```
So task IDs would work, but any number that's not a task ID would be used for search or to create a new task.